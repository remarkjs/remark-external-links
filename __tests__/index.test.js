/* eslint-env jest */

var remark = require('remark')
var html = require('remark-html')
var externalLinks = require('..')

var input = [
  '[remark](https://github.com/remarkjs/remark)',
  '',
  '[relative link](./example.md) and [fragment link](#fragment)',
  '',
  '[missing][], [local][], and [external][].',
  '',
  '[current](.) [up](..) [relative link without ./](example.md)',
  '',
  '[local]: #local',
  '[external]: https://github.com/remarkjs/remark'
].join('\n')

test('should add the defaults when without options', function() {
  expect(
    remark()
      .use(externalLinks)
      .use(html)
      .processSync(input)
      .toString()
  ).toMatchSnapshot()
})

test('should do nothing if both are set to false', function() {
  expect(
    remark()
      .use(externalLinks, {target: false, rel: false})
      .use(html)
      .processSync(input)
      .toString()
  ).toMatchSnapshot()
})

test('should add a target', function() {
  expect(
    remark()
      .use(externalLinks, {target: '_blank', rel: false})
      .use(html)
      .processSync(input)
      .toString()
  ).toMatchSnapshot()
})

test('should add a rel', function() {
  expect(
    remark()
      .use(externalLinks, {target: false, rel: 'nofollow'})
      .use(html)
      .processSync(input)
      .toString()
  ).toMatchSnapshot()
})

test('should add both', function() {
  expect(
    remark()
      .use(externalLinks, {target: '_blank', rel: 'nofollow'})
      .use(html)
      .processSync(input)
      .toString()
  ).toMatchSnapshot()
})
