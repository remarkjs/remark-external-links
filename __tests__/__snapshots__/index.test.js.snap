// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a rel 1`] = `
"<p><a href=\\"https://github.com/remarkjs/remark\\" rel=\\"nofollow\\">remark</a></p>
<p><a href=\\"./example.md\\">relative link</a> and <a href=\\"#fragment\\">fragment link</a></p>
<p>[missing][], <a href=\\"#local\\">local</a>, and <a href=\\"https://github.com/remarkjs/remark\\" rel=\\"nofollow\\">external</a>.</p>
<p><a href=\\".\\">current</a> <a href=\\"..\\">up</a> <a href=\\"example.md\\">relative link without ./</a></p>
"
`;

exports[`should add a target 1`] = `
"<p><a href=\\"https://github.com/remarkjs/remark\\" target=\\"_blank\\">remark</a></p>
<p><a href=\\"./example.md\\">relative link</a> and <a href=\\"#fragment\\">fragment link</a></p>
<p>[missing][], <a href=\\"#local\\">local</a>, and <a href=\\"https://github.com/remarkjs/remark\\" target=\\"_blank\\">external</a>.</p>
<p><a href=\\".\\">current</a> <a href=\\"..\\">up</a> <a href=\\"example.md\\">relative link without ./</a></p>
"
`;

exports[`should add both 1`] = `
"<p><a href=\\"https://github.com/remarkjs/remark\\" target=\\"_blank\\" rel=\\"nofollow\\">remark</a></p>
<p><a href=\\"./example.md\\">relative link</a> and <a href=\\"#fragment\\">fragment link</a></p>
<p>[missing][], <a href=\\"#local\\">local</a>, and <a href=\\"https://github.com/remarkjs/remark\\" target=\\"_blank\\" rel=\\"nofollow\\">external</a>.</p>
<p><a href=\\".\\">current</a> <a href=\\"..\\">up</a> <a href=\\"example.md\\">relative link without ./</a></p>
"
`;

exports[`should add the defaults when without options 1`] = `
"<p><a href=\\"https://github.com/remarkjs/remark\\" target=\\"_blank\\" rel=\\"nofollow noopener noreferrer\\">remark</a></p>
<p><a href=\\"./example.md\\">relative link</a> and <a href=\\"#fragment\\">fragment link</a></p>
<p>[missing][], <a href=\\"#local\\">local</a>, and <a href=\\"https://github.com/remarkjs/remark\\" target=\\"_blank\\" rel=\\"nofollow noopener noreferrer\\">external</a>.</p>
<p><a href=\\".\\">current</a> <a href=\\"..\\">up</a> <a href=\\"example.md\\">relative link without ./</a></p>
"
`;

exports[`should do nothing if both are set to false 1`] = `
"<p><a href=\\"https://github.com/remarkjs/remark\\">remark</a></p>
<p><a href=\\"./example.md\\">relative link</a> and <a href=\\"#fragment\\">fragment link</a></p>
<p>[missing][], <a href=\\"#local\\">local</a>, and <a href=\\"https://github.com/remarkjs/remark\\">external</a>.</p>
<p><a href=\\".\\">current</a> <a href=\\"..\\">up</a> <a href=\\"example.md\\">relative link without ./</a></p>
"
`;
